samplekotlin/src/main/java/com/example/chattutorial/ChannelActivity.kt

```diff
@@ -3,32 +3,31 @@ package com.example.chattutorial
 import android.content.Context
 import android.content.Intent
 import android.os.Bundle
-import androidx.activity.OnBackPressedCallback
+import androidx.activity.addCallback
+import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
-import androidx.lifecycle.ViewModelProvider
 import com.getstream.sdk.chat.viewmodel.ChannelHeaderViewModel
 import com.getstream.sdk.chat.viewmodel.MessageInputViewModel
 import com.getstream.sdk.chat.viewmodel.bindView
 import com.getstream.sdk.chat.viewmodel.factory.ChannelViewModelFactory
 import com.getstream.sdk.chat.viewmodel.messages.MessageListViewModel
 import com.getstream.sdk.chat.viewmodel.messages.bindView
 import io.getstream.chat.android.client.models.Channel
-import kotlinx.android.synthetic.main.activity_channel.channelHeaderView
-import kotlinx.android.synthetic.main.activity_channel.messageInputView
-import kotlinx.android.synthetic.main.activity_channel.messageListView
-
+import kotlinx.android.synthetic.main.activity_channel.*
 
 class ChannelActivity : AppCompatActivity(R.layout.activity_channel) {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        val cid = checkNotNull(intent.getStringExtra(CID_KEY)) {"Specifying a channel id is required when starting ChannelActivity"}
+        val cid = checkNotNull(intent.getStringExtra(CID_KEY)) {
+            "Specifying a channel id is required when starting ChannelActivity"
+        }
 
         // step 1 - we create 3 separate ViewModels for the views so it's easy to customize one of the components
-        val viewModelProvider = ViewModelProvider(this, ChannelViewModelFactory(cid))
-        val channelHeaderViewModel = viewModelProvider.get(ChannelHeaderViewModel::class.java)
-        val messageListViewModel = viewModelProvider.get(MessageListViewModel::class.java)
-        val messageInputViewModel = viewModelProvider.get(MessageInputViewModel::class.java)
+        val factory = ChannelViewModelFactory(cid)
+        val channelHeaderViewModel: ChannelHeaderViewModel by viewModels { factory }
+        val messageListViewModel: MessageListViewModel by viewModels { factory }
+        val messageInputViewModel: MessageInputViewModel by viewModels { factory }
```

Comment:
Seems to me that obtaining a view model instance is standard enough that we can use a cleaner, lazy delegate instead of doing it long-form. The tutorial doesn't really care about _how_ you obtain your VMs, just that you do.

Reply:
Agreed. I was surprised by this at first, but apparently a lot of people are using this syntax now, as Google pushes this in their sample code all the time. Many new devs won't even know about the `ViewModelProvider` types at this point.